frontend/src/components/elements/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,176 @@
+/**
+ * @license
+ * Copyright 2018-2022 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ComponentType, ReactElement, useRef, useState } from "react"
+import { useTheme } from "@emotion/react"
+import { Tabs as UITabs, Tab as UITab } from "baseui/tabs-motion"
+
+import { BlockNode, AppNode } from "src/lib/AppNode"
+import { useIsOverflowing } from "src/lib/Hooks"
+
+import { StyledTabContainer } from "./styled-components"
+
+const TAB_HEIGHT = "2.5rem"
+
+export interface Props {
+  widgetsDisabled: boolean
+  node: BlockNode
+  isStale: boolean
+}
+
+function Tabs(TabLayoutComponent: ComponentType<any>): ComponentType<any> {
+  const TabContainer = (props: Props): ReactElement => {
+    const { widgetsDisabled, node, isStale } = props
+
+    const [activeKey, setActiveKey] = useState("0")
```

Comment:
it there a reason this is intentionally / carefully kept as a string rather than an integer index? Would be worth adding a comment explaining that BaseWeb expects this or whatever the reason is